### PR TITLE
Fix/native/passphrase flow errors

### LINKED ISSUE #13190: Passphrase duplicate & mismatch
Passphrase duplicate & mismatch in the flow of creating new wallet behave unpredictable (navigation jumps). My suspicion is that we react to error that is set in redux, but resetting the state is not correct.

### PR DESCRIPTION
**Pull request by juriczech**
## Description
Refactor redux logic of passphrase flow so that the error state is reset and edge cases are handled on appropriate screens. 

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/13190

## Screenshots:
https://satoshilabs.slack.com/archives/C03H2Q5MX51/p1720779497180219

### CODE REVIEW

File: suite-native/device-authorization/src/deviceAuthorizationSlice.ts [hunk 1]
```diff
@@ -10,14 +10,20 @@ import {
...
 type DeviceAuthorizationState = {
     hasDeviceRequestedPin: boolean;
     hasDeviceRequestedPassphrase: boolean;
-    passphraseError: AuthorizeDeviceError | CreateDeviceInstanceError | null;
+    passphraseError:
+        | AuthorizeDeviceError
+        | CreateDeviceInstanceError
+        | VerifyPassphraseOnEmptyWalletError
+        | null;
     isVefifyingPassphraseOnEmptyWallet: boolean;
```
Comment (vytick) [inline suite-native/device-authorization/src/deviceAuthorizationSlice.ts]: Just noticed typo

```suggestion
    isVerifyingPassphraseOnEmptyWallet: boolean;
```

File: suite-native/device-authorization/src/deviceAuthorizationSlice.ts [hunk 2]
```diff
@@ -118,18 +139,27 @@ export const selectDeviceRequestedAuthorization = (state: DeviceAuthorizationRoo
...
+
 export const selectPassphraseDuplicateError = (state: DeviceAuthorizationRootState) => {
-    return state.deviceAuthorization.passphraseError?.error === 'passphrase-duplicate'
-        ? state.deviceAuthorization.passphraseError
+    return selectPassphraseError(state)?.error === 'passphrase-duplicate'
+        ? (state.deviceAuthorization.passphraseError as AuthorizeDeviceError)
         : null;
 };
 
+export const selectHasVerifyCancelledError = (state: DeviceAuthorizationRootState) =>
```
Comment (vytick) [inline suite-native/device-authorization/src/deviceAuthorizationSlice.ts]:
```suggestion
export const selectHasVerificationCancelledError = (state: DeviceAuthorizationRootState) =>
```

File: suite-native/module-authorize-device/src/screens/passphrase/PassphraseConfirmOnTrezorScreen.tsx [hunk 3]
```diff
@@ -53,6 +66,44 @@ export const PassphraseConfirmOnTrezorScreen = () => {
         }
     }, [device, dispatch, isDeviceConnectedAndAuthorized, isDiscoveryActive, navigation]);
 
+    useEffect(() => {
+        // User has canceled the authorization process on device (authorizeDeviceThunk rejects with auth-failed error)
+        if (hasAuthorizationFailed || hasVerifyCancelledError) {
```
Comment (vytick) [inline suite-native/module-authorize-device/src/screens/passphrase/PassphraseConfirmOnTrezorScreen.tsx]: Should canceling on device when confirming passphrase second time cancel whole flow?

Comment (juriczech) [reply suite-native/module-authorize-device/src/screens/passphrase/PassphraseConfirmOnTrezorScreen.tsx]: I believe so

Comment (juriczech) [reply suite-native/module-authorize-device/src/screens/passphrase/PassphraseConfirmOnTrezorScreen.tsx]: Just to clarify - it's the same as the close button on the screen in app, Trezor device just doesn't display warning sheet, so we kill the flow.

File: suite-native/device-authorization/src/deviceAuthorizationSlice.ts (hunk 1 above)
Comment (juriczech) [reply suite-native/device-authorization/src/deviceAuthorizationSlice.ts]: These make me giggle. At least the selector was correct https://github.com/trezor/trezor-suite/pull/13246/commits/c7515cb0f2ab518414a840eb99d21b9562dbfae4

File: suite-native/device-authorization/src/deviceAuthorizationSlice.ts (hunk 2 above)
Comment (juriczech) [reply suite-native/device-authorization/src/deviceAuthorizationSlice.ts]: https://github.com/trezor/trezor-suite/pull/13246/commits/c7515cb0f2ab518414a840eb99d21b9562dbfae4

**Review by vytick — APPROVED** [review]
LGTM 🚀  good job!

**Pull request merged**